### PR TITLE
Implemented the syncing between the NavMeshAgent and bot positions

### PR DESCRIPTION
**Pull request by nfetissow**
Created a separate MonoBehaviour to control the synchronisation of the bot's transform position with Unity NavMeshAgent position. Now the bots avoid collisions and find paths much better than before.

### CODE REVIEW

**Comment by EnigmaticDragon** [comment]
> Currently the bots are not able to attack the boss for me - does it work for you? Maybe the range needs to be increased.

Happens for me, too. When they are aggressive and run to the boss, they just end up standing in front of it, doing nothing.

**Comment by nfetissow** [comment]
Should work on Game.unity. 

Errors on _botsDevelop don't matter. 
Also, the proper distance calculation is what I'll be working on next, but it is very much unrelated to changes in this PR, so I'll make a new branch for it.

File: Assets/Scripts/Bots/AI/NavMeshAgentDriver.cs [hunk 1]
```diff
@@ -80,7 +87,7 @@ public bool ReachedDestination()
         public void SetDestination(Vector3 desiredWorldPosition)
         {
             desiredWorldPosition.y = 0f;
-            agent.nextPosition = transform.position + (desiredWorldPosition - transform.position).normalized * agent.radius;
+            agent.Warp(transform.position + (desiredWorldPosition - transform.position).normalized * agent.radius);
```
Comment (lee17017) [inline Assets/Scripts/Bots/AI/NavMeshAgentDriver.cs]: Why did you change this to warp? Now the movement of the bots are blocky when they are running. And they continue to move when they are attacking.

Comment (nfetissow) [reply Assets/Scripts/Bots/AI/NavMeshAgentDriver.cs]: I wasn't aware that the Warp method will update the position of the transform as well. Should be fixed now.

**Comment by nfetissow** [comment]
I fixed the movement of aggressive behaviour.
The non-reaching of the boss for the melee attacks will be fixed in a separate PR
